pandas/plotting/_core.py

```diff
@@ -370,1305 +642,760 @@
-            :meth:`pandas.Series.plot`.
+            - A column name or position whose values will be used to color the
+              marker points according to a colormap.
+
+        **kwds
+            Keyword arguments to pass on to :meth:`DataFrame.plot`.
 
         Returns
         -------
         :class:`matplotlib.axes.Axes` or numpy.ndarray of them
-        """
-        return self(kind='hist', bins=bins, **kwds)
-
-    @Appender(_shared_docs['kde'] % {
-        'this-datatype': 'Series',
-        'sibling-datatype': 'DataFrame',
-        'examples': """
-        Given a Series of points randomly sampled from an unknown
-        distribution, estimate its PDF using KDE with automatic
-        bandwidth determination and plot the results, evaluating them at
-        1000 equally spaced points (default):
-
-        .. plot::
-            :context: close-figs
-
-            >>> s = pd.Series([1, 2, 2.5, 3, 3.5, 4, 5])
-            >>> ax = s.plot.kde()
-
-        A scalar bandwidth can be specified. Using a small bandwidth value can
-        lead to over-fitting, while using a large bandwidth value may result
-        in under-fitting:
 
-        .. plot::
-            :context: close-figs
+        See Also
+        --------
+        matplotlib.pyplot.scatter : Scatter plot using multiple input data
+            formats.
 
-            >>> ax = s.plot.kde(bw_method=0.3)
+        Examples
+        --------
+        Let's see how to draw a scatter plot using coordinates from the values
+        in a DataFrame's columns.
 
         .. plot::
             :context: close-figs
 
-            >>> ax = s.plot.kde(bw_method=3)
+            >>> df = pd.DataFrame([[5.1, 3.5, 0], [4.9, 3.0, 0], [7.0, 3.2, 1],
+            ...                    [6.4, 3.2, 1], [5.9, 3.0, 2]],
+            ...                   columns=['length', 'width', 'species'])
+            >>> ax1 = df.plot.scatter(x='length',
+            ...                       y='width',
+            ...                       c='DarkBlue')
 
-        Finally, the `ind` parameter determines the evaluation points for the
-        plot of the estimated PDF:
+        And now with the color determined by a column as well.
 
         .. plot::
             :context: close-figs
 
-            >>> ax = s.plot.kde(ind=[1, 2, 3, 4, 5])
-        """.strip()
-    })
-    def kde(self, bw_method=None, ind=None, **kwds):
-        return self(kind='kde', bw_method=bw_method, ind=ind, **kwds)
-
-    density = kde
-
-    def area(self, **kwds):
+            >>> ax2 = df.plot.scatter(x='length',
+            ...                       y='width',
+            ...                       c='species',
+            ...                       colormap='viridis')
         """
-        Area plot.
+        if not isinstance(self._parent,
+                          pandas.core.dtypes.generic.ABCDataFrame):
+            raise ValueError(
+                "plot kind scatter can only be used for data frames")
```

Comment:
Doesn't this duplicate the check at https://github.com/pandas-dev/pandas/pull/27009/files#diff-2b118bda866f3d626ceb6b529c62cd1aR222 ?